test/basic.js

```diff
@@ -128,6 +130,31 @@ t.test('--harmony', function (t) {
   })
 })
 
+t.test('node exe with different name', function(t) {
+  var fp = path.join(__dirname, 'fixtures', 'exething')
+  fs.createReadStream(process.execPath)
```

Comment:
Yes, that's why `fs.symlink` won't work.  But `fs.link` creates a hard link, not a symlink.